api/namex/models/request.py

```diff
@@ -331,11 +331,14 @@ def find_existing_name_by_user(cls, user_name_search_string, email):
         """	
         Gets requests submited by user with given name choice in state draft
         """
+        current_time = datetime.utcnow()
         existing_nr = db.session.query(Request). \
             join(Applicant, and_(Applicant.nrId == Request.id)). \
             filter(
                 Applicant.emailAddress == email,
                 (Request.stateCd == 'DRAFT') | (Request.stateCd == 'PENDING_PAYMENT') ,
+                #check if the request is being made within 5 minutes, if time >= 5 minutes, it will let the user create a new request with same name otherwise it will give the error message to prevent accidental duplicates and accidental double payments for them.
+                Request.submittedDate >= current_time - timedelta(minutes=5),
                 (Request.nameSearch == ('('+user_name_search_string+')')) | ( Request.nameSearch == user_name_search_string )). \
             one_or_none()
```

Comment:
also, this method throws an exception if more than one row is returned: https://docs.sqlalchemy.org/en/14/orm/query.html#sqlalchemy.orm.Query.one_or_none - maybe i don't understand the logic, but if we actually get multiple rows, would not this result in an unhandled exception

Reply:
yes please fix that exception, Rajan

Reply:
updated code to check on only for pending payment state if request stuck for payment about 5 mins or more then user can send in another request, once nr is fully completed with payment then system will not allow duplicates not even intentional ones.

Reply:
looks good to me